### PR TITLE
Tox heal chems diversified

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Diversifies toxin-healing chems

## Why It's Good For The Game

Someone else fucked things up and I wanted to fix it.

We don't need 3 chems doing the exact same thing, so now we don't.

Arithrazine is reset to exactly what it was before, as it's a component of some pre-mixed medicines.
Hyronalin is changed to the T2 version of antitox. It heals 1 toxin/tick with no side effects, but doesn't purge hostile chems.

Ryetalin is changed to a long-duration shield against toxins, but heals no damage on it's own. A 30u dose will last for ~40 minutes, but will only last 60 seconds or less if mixed with Tricord, Bicard, or Kelo.

Russian Red is changed to an emergency revival medicine like Neuraline. 1u/tick metabolism rate, 5 bruteheal, 5 burnheal, 2.5 toxheal, but does 2 cloneloss. 

This can only be healed in cryo.
Pills are 10u each, so you'll heal 50-125 damage for 20 permanent damage. And as an added bonus, you don't even lose the use of your limb to mutation.

Added a pill packet for Ryetalyn and Russian Red.
Replaced one of the 3 duplicate hypervene autoinjectors in Toxin First Aid Kits with a Ryetalyn pill packet.
Replaced the awkward and slow-acting syringe kit in the Normal First Aid Kit with a Russian Red pill packet.

Altered 2 pill's descriptions, and added descriptions to most pill bottles.

## Changelog
:cl:
balance: Rebalanced toxheal chems
spellcheck: Fixed a bad copy/paste on a pill description
/:cl:

### CODE REVIEW

**Comment by jroinc** [comment]
Tested, functions as expected.

**Comment by Walarks** [comment]
I think Ryetalin is just a more unbalanced version of Dylovene. It seems like it will be a staple in any miracle pill combination to fully counter all negative reagents. 

I think giving it a short duration to be taken reactively or making it purge meds that heal damage in addition to toxins would be better as it makes you pick which one to take based on what you are faced with.

**Comment by jroinc** [comment]
> I think Ryetalin is just a more unbalanced version of Dylovene. It seems like it will be a staple in any miracle pill combination to fully counter all negative reagents.
> 
> I think giving it a short duration to be taken reactively or making it purge meds that heal damage in addition to toxins would be better as it makes you pick which one to take based on what you are faced with.

I can't give it a differential purge rate, so that's out. I could make it react with healchems, but that's annoying code. Easy, just some duplication.

I might give it a 20% chance of 0.5 stamina damage so it horribly cripples run mode. Thoughts on that?

**Comment by Walarks** [comment]
> > I think Ryetalin is just a more unbalanced version of Dylovene. It seems like it will be a staple in any miracle pill combination to fully counter all negative reagents.
> > I think giving it a short duration to be taken reactively or making it purge meds that heal damage in addition to toxins would be better as it makes you pick which one to take based on what you are faced with.
> 
> I can't give it a differential purge rate, so that's out. I could make it react with healchems, but that's annoying code. Easy, just some duplication.
> 
> I might give it a 20% chance of 0.5 stamina damage so it horribly cripples run mode. Thoughts on that?

I think the main issue is that its metabolism is too low for it to effectively nullify any negative reagents. Maybe it can have the metabolism doubled and have it deal stamina damage/higher % to deal stamina damage based on how many others meds you are using.

**Review by TiviPlus — COMMENTED** [review]
Agree with slotbot here, other chems are good though. Duke does stamina iirc but you could maybe make it purge non toxins chems?

**Comment by jroinc** [comment]
> I think the main issue is that its metabolism is too low for it to effectively nullify any negative reagents. Maybe it can have the metabolism doubled and have it deal stamina damage/higher % to deal stamina damage based on how many others meds you are using.

Metabolism has no effects on it's effect. Metabolism is only responsible for how much of the indicated reagent is removed per tick. Whether it's 1000u or 0.0001u, it'll still have the effects defined in on mob life, then remove that much. The 0.0001u one will last much longer, that's all.

The defined effects are that it removes 5u/tick of other poison reagents. Low metabolism just means that it'll stick around forever. My toxins have medpurge so multiple stings remove the protection factor, but yours don't, so I'll throw in another balancing factor.


> Agree with slotbot here, other chems are good though. Duke does stamina iirc but you could maybe make it purge non toxins chems?

20% chance of 1 stamina damage is gonna kill your ability to run (not sure if I wanna do that yet). Now, I can't do selective purge rates (so I can't have it purge non-toxins). I could have it REACT with select non-toxins, but that's clunky. I think the best idea is to put a 1u/tick purge on tricord, bicard, and kelo. 

That way, it'll last basically forever until you take damage and start poppin' pills, which will gradually bleed your protection away.

**Comment by jroinc** [comment]
Tested, functional. No further changes this PR unless asked.

File: code/game/objects/items/storage/pill_packets.dm [hunk 1]
```diff
@@ -11,16 +11,26 @@
...
 	desc = "This packet containts paracetamol pills, also known as tylenol. A long lasting but minor painkiller. Once you take them out they don't go back in. No more than 4 pills in a long period."
 	pill_type_to_fill = /obj/item/reagent_containers/pill/paracetamol
 
-
 /obj/item/storage/pill_bottle/packet/leporazine
 	name = "Leporazine pill packet"
 	desc = "This packet containts leporazine pills. Rapidly stablizes the patients internal temperature. Once you take them out they don't go back in.."
 	pill_type_to_fill = /obj/item/reagent_containers/pill/leporazine
+
+/obj/item/storage/pill_bottle/packet/russianRed
```
Comment (TiviPlus) [inline code/game/objects/items/storage/pill_packets.dm:27]: Snake case

Comment (jroinc) [reply code/game/objects/items/storage/pill_packets.dm:27]: Uses the same case as the pre-existing chem, pill, and pill bottle for consistency. I can change the things I added if you really want me to, but that seems awkward.

Comment (TiviPlus) [reply code/game/objects/items/storage/pill_packets.dm:27]: snake case

Comment (jroinc) [reply code/game/objects/items/storage/pill_packets.dm:27]: Yes, that's a statement. Do you want snake case, or do you want me to use a different typepath than the other things?

File: code/game/objects/items/reagent_containers/pill.dm [hunk 2]
```diff
@@ -178,10 +178,15 @@
 	pill_id = 14
 
 /obj/item/reagent_containers/pill/russianRed
```
Comment (Hughgent) [inline code/game/objects/items/reagent_containers/pill.dm:180]:
```suggestion
/obj/item/reagent_containers/pill/russian_red
```

File: code/game/objects/items/storage/firstaid.dm [hunk 3]
```diff
@@ -364,16 +367,19 @@
...
 
 /obj/item/storage/pill_bottle/russianRed
 	name = "\improper Russian Red pill bottle"
+	desc = "Contains pills that heal all damage rapidly at the cost of small amounts of unhealable damage."
 	icon_state = "pill_canister1"
 	pill_type_to_fill = /obj/item/reagent_containers/pill/russianRed
 
 /obj/item/storage/pill_bottle/quickclot
 	name = "quick-clot pill bottle"
+	desc = "Contains pills that suppress IB while waiting for a full treatment."
```
Comment (Hughgent) [inline code/game/objects/items/storage/firstaid.dm:382]:
```suggestion
	desc = "Contains pills that suppress internal bleeding while waiting for full treatment."
```

File: code/game/objects/items/storage/pill_packets.dm (hunk 1 above)
Comment (Hughgent) [reply code/game/objects/items/storage/pill_packets.dm:27]:
```suggestion
/obj/item/storage/pill_bottle/packet/russian_red
```

File: code/game/objects/items/storage/pill_packets.dm [hunk 4]
```diff
@@ -11,16 +11,26 @@
...
+	name = "Russian Red pill packet"
+	desc = "This packet containts Russian Red pills. Used for field treatment of critical cases without a medic. Once you take them out they don't go back in.."
+	pill_type_to_fill = /obj/item/reagent_containers/pill/russianRed
```
Comment (Hughgent) [inline code/game/objects/items/storage/pill_packets.dm:30]:
```suggestion
	pill_type_to_fill = /obj/item/reagent_containers/pill/russian_red
```
If this is contained anywhere else in code, adjust accordingly.

File: code/modules/reagents/reagents/medical.dm [hunk 5]
```diff
@@ -482,20 +488,17 @@
 
 /datum/reagent/medicine/russianred
```
Comment (Hughgent) [inline code/modules/reagents/reagents/medical.dm:489]:
```suggestion
/datum/reagent/medicine/russian_red
```
gotta catch em all.

**Review by Hughgent — CHANGES_REQUESTED** [review]
I'm ok with the changes. be prepared to update the wiki though.

**Comment by jroinc** [comment]
Updated, no errors far as I can tell.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**